src/dbus/dbus_thread_object.cpp

```diff
@@ -0,0 +1,558 @@
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_SUPPORTED_CHANNEL_MASK_PROPERTY,
+                               std::bind(&DBusThreadObject::GetSupportedChannelMaskHandler, this, _1));
+
+    return err;
+}
+
+void DBusThreadObject::DeviceRoleHandler(otDeviceRole aDeviceRole)
+{
+    SignalPropertyChanged(OTBR_DBUS_PREFIX, OTBR_DBUS_DEVICE_ROLE_PROPERTY, GetDeviceRoleName(aDeviceRole));
+}
+
+void DBusThreadObject::ScanHandler(DBusRequest &aRequest)
+{
+    auto threadHelper = mNcp->GetThreadHelper();
+    threadHelper->Scan(std::bind(&DBusThreadObject::ReplyScanResult, this, aRequest, _1, _2));
+}
+
+void DBusThreadObject::ReplyScanResult(DBusRequest &                          aRequest,
+                                       otError                                aError,
+                                       const std::vector<otActiveScanResult> &aResult)
+{
+    std::vector<otbrActiveScanResult> results;
+
+    if (aError != OT_ERROR_NONE)
+    {
+        aRequest.ReplyOtbrResult(static_cast<otbrError>(aError));
+    }
+    else
+    {
+        for (const auto &r : aResult)
+        {
+            otbrActiveScanResult result;
+
+            memcpy(&result.mExtAddress, r.mExtAddress.m8, sizeof(result.mExtAddress));
+            memcpy(&result.mExtendedPanId, r.mExtendedPanId.m8, sizeof(result.mExtendedPanId));
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+            result.mExtAddress    = bswap_64(result.mExtAddress);
+            result.mExtendedPanId = bswap_64(result.mExtendedPanId);
+#endif
+            result.mNetworkName = r.mNetworkName.m8;
+            result.mSteeringData =
+                std::vector<uint8_t>(r.mSteeringData.m8, r.mSteeringData.m8 + r.mSteeringData.mLength);
+            result.mPanId         = r.mPanId;
+            result.mJoinerUdpPort = r.mJoinerUdpPort;
+            result.mChannel       = r.mChannel;
+            result.mRssi          = r.mRssi;
+            result.mLqi           = r.mLqi;
+            result.mVersion       = r.mVersion;
+            result.mIsNative      = r.mIsNative;
+            result.mIsJoinable    = r.mIsJoinable;
+
+            results.emplace_back(result);
+        }
+
+        aRequest.Reply(std::tie(results));
+    }
+}
+
+void DBusThreadObject::AttachHandler(DBusRequest &aRequest)
+{
+    auto                 threadHelper = mNcp->GetThreadHelper();
+    std::string          name;
+    uint16_t             panid;
+    uint64_t             extPanId;
+    std::vector<uint8_t> masterKey;
+    std::vector<uint8_t> pskc;
+    uint32_t             channelMask;
+
+    auto args = std::tie(masterKey, panid, name, extPanId, pskc, channelMask);
+
+    if (DBusMessageToTuple(*aRequest.GetMessage(), args) != OTBR_ERROR_NONE)
+    {
+        aRequest.ReplyOtbrResult(OTBR_OT_ERROR_INVALID_ARGS);
+    }
+    else
+    {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+        extPanId = bswap_64(extPanId);
```

Comment:
`DBusMessageToTuple(*aRequest.GetMessage(), args)` indicates that `extPanId` is encoded as `uint64_t`, then D-Bus APIs should have handled endianness for us, why `bswap` here?

Reply:
instead of swapping here, we should do conversion inside `ThreadHelper::Attach(...)`. Otherwise parameter `aExtPanId` of `ThreadHelper::Attach` is in corrupted status. Agree?

Reply:
Keep it consistent that each `uint64_t` variable is not in corrupted status.